### PR TITLE
Allow user to specify expansion for dropdown lists in site navigation

### LINKED ISSUE #348: SiteNav: allow specifying an expansion level
v1.9.0

Current: All drop-down items are collapsed by default

Suggestion: we can allow specifying the default expansion level as an attribute

### PR DESCRIPTION
**Pull request by Chng-Zhi-Xuan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Enhancement to an existing feature

Resolves #348 

**What is the rationale for this request?**
User would like to expand certain dropdown lists in site navigation by default.

**What changes did you make? (Give an overview)**
- Add alternative formatting logic for dropdown lists when detecting keyword in the title
- Update site-nav test within `test_site`
- Update Site navigation section in user guide

Before | After
--- | ---
348-before | 348-after

**Is there anything you'd like reviewers to focus on?**
- Variable naming
- Wording in user guide

**Testing instructions:**
- Checkout this PR and go to the `test_site` folder
- Run `markbind serve` and check that certain dropdown lists are expanded by default
- Modify `site-nav.md` and see the corresponding changes reflected after running `markbind serve` again

### CODE REVIEW

**Comment by yamgent** [comment]
If I want to use the text `expanded="true"` in my navbar item, I can no longer do so:

```markdown
<navigation>
* Nested Dropdown title :triangular_ruler: expanded="true" <!-- expected: will expand -->
    * [**Nested** Dropdown link one](https://www.google.com/)
    * [**Nested** Dropdown link two](https://www.google.com/)
* Nested Dropdown title :triangular_ruler: expanded=&quot;true&quot; <!-- expected: rendered as text -->
    * [**Nested** Dropdown link one](https://www.google.com/)
    * [**Nested** Dropdown link two](https://www.google.com/)
</navigation>
```

Expected:



Actual:

**Comment by Chng-Zhi-Xuan** [comment]
**Key issue**

`md.render(...)` will convert any special html symbols in Markdown to their respective HTML character codes. E.G `"` will become `&quot;` , so both statements shown above will be identical when getting processed in the `formatSiteNav` function.

File: docs/userGuide/contentAuthoring.md [hunk 1]
```diff
@@ -435,14 +435,15 @@ If you would like to have a site navigation bar for your site, you may create on
     - Ensure you wrap your navigation layout in a `<navigation>` tag for MarkBind to render it properly. Also, there should only be one `<navigation>` tag in your file.
     - You may author additional HTML and Markdown content outside the `<navigation>` tag.
     - The `<navigation>` tag is also optional, allowing you to author everything in HTML and Markdown syntax.
+    - Use `:expanded:` keyword at the end of Dropdown titles to expand them by default.
```
Comment (yamgent) [inline docs/userGuide/contentAuthoring.md]: This is not strictly true. The current algorithm allows adding `:expanded:` at **any** location in the Dropdown title, not just at the end only.

File: lib/Page.js [hunk 2]
```diff
@@ -136,11 +137,24 @@ function formatSiteNav(renderedSiteNav) {
...
       // Replace the title with the dropdown wrapper
       dropdownTitle.remove();
-      nestedList.wrap('<div class="dropdown-container"></div>');
-      $(this).prepend('<button class="dropdown-btn">'
-        + `${dropdownTitle} `
-        + `${DROPDOWN_BUTTON_ICON_HTML}\n`
-        + '</button>');
+      // Check if dropdown is expanded by default
+      if (dropdownTitle.toString().includes(DROPDOWN_EXPAND_KEYWORD)) {
+        const keywordRegex = new RegExp(DROPDOWN_EXPAND_KEYWORD, 'g');
```
Comment (yamgent) [inline lib/Page.js]: `expandKeywordRegex`?

File: lib/Page.js [hunk 3]
```diff
@@ -136,11 +137,24 @@ function formatSiteNav(renderedSiteNav) {
...
+        const dropdownTitleNoKeyword = dropdownTitle.toString().replace(keywordRegex, '');
```
Comment (yamgent) [inline lib/Page.js]: `dropdownTitleWithoutKeyword`?

File: lib/Page.js [hunk 4]
```diff
@@ -136,11 +137,24 @@ function formatSiteNav(renderedSiteNav) {
...
+        const iconSelector = cheerio.load(DROPDOWN_BUTTON_ICON_HTML);
+        const rotatedIconHtml = iconSelector('i').addClass('rotate-icon');
```
Comment (yamgent) [inline lib/Page.js]: Seems like you can just compress the above two lines into just one.

```javascript
const rotatedIconHtml = cheerio.load(DROPDOWN_BUTTON_ICON_HTML)('i').addClass('rotate-icon');
```

File: docs/userGuide/contentAuthoring.md (hunk 1 above)
Comment (yamgent) [reply docs/userGuide/contentAuthoring.md]: Maybe create a test case for one that does not have `:expanded:` at the back? All of them currently have the keyword at the back at least.

File: lib/Page.js (hunk 4 above)
Comment (Chng-Zhi-Xuan) [reply lib/Page.js]: Had to add `cheerio.load( ... , { xmlMode: false })` , else the `.toString()` and `.html()` will produce self closing tags (not supported in HTML5)

**Comment by Chng-Zhi-Xuan** [comment]
**Update**
- Squashed commits

**Pull request merged**